tests/Doctrine/Tests/DBAL/Functional/ModifyLimitQueryTest.php

```diff
@@ -110,6 +110,20 @@ public function testModifyLimitQuerySubSelect()
         $this->assertLimitResult(array(2, 1), $sql, 2, 2, false);
     }
 
+    public function testModifyLimitQueryFromSubSelect()
+    {
+        $this->_conn->insert('modify_limit_table', array('test_int' => 1));
+        $this->_conn->insert('modify_limit_table', array('test_int' => 2));
+        $this->_conn->insert('modify_limit_table', array('test_int' => 3));
+        $this->_conn->insert('modify_limit_table', array('test_int' => 4));
+
+        $sql = "SELECT * FROM (SELECT * FROM modify_limit_table) sub";
+
+        $this->assertLimitResult(array(4, 3, 2, 1), $sql, 10, 0, false);
```

Comment:
No, no, no this doesn't make sense without and order by. While I appreciate your efforts, this is just plain undeterministic. I know the "old" tests are not the best, but please let's not introduce more....